clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/AbstractVeniceMapper.java

```diff
@@ -97,25 +108,67 @@ protected boolean process(
     if (recordKey == null) {
       throw new VeniceException("Mapper received a empty key record");
     }
+
     if (recordValue == null) {
       LOGGER.warn("Received null record, skip.");
       MRJobCounterHelper.incrEmptyRecordCount(reporter, 1);
       return false;
     }
+
+    // both key and value are not null
+    MRJobCounterHelper.incrTotalKeySize(reporter, recordKey.length);
     MRJobCounterHelper.incrTotalUncompressedValueSize(reporter, recordValue.length);
 
+    // Compress and save the details based on the configured compression strategy: This should not fail
+    byte[] finalRecordValue;
     try {
-      recordValue = compressor.compress(recordValue);
+      finalRecordValue = compressor[compressionStrategy.getValue()].compress(recordValue);
     } catch (IOException e) {
       throw new VeniceException(
           "Caught an IO exception while trying to to use compression strategy: "
-              + compressor.getCompressionStrategy().name(),
+              + compressor[compressionStrategy.getValue()].getCompressionStrategy().name(),
           e);
     }
-    MRJobCounterHelper.incrTotalKeySize(reporter, recordKey.length);
-    MRJobCounterHelper.incrTotalValueSize(reporter, recordValue.length);
+    // record the final stored value details
+    MRJobCounterHelper.incrTotalValueSize(reporter, finalRecordValue.length);
     keyBW.set(recordKey, 0, recordKey.length);
-    valueBW.set(recordValue, 0, recordValue.length);
+    valueBW.set(finalRecordValue, 0, finalRecordValue.length);
+
+    // Compress based on all compression strategies to collect metrics if enabled.
+    if (compressionMetricCollectionEnabled) {
+      byte[] compressedRecordValue;
+      for (CompressionStrategy compressionStrategy: CompressionStrategy.values()) {
+        if (compressor[compressionStrategy.getValue()] != null) {
+          switch (compressionStrategy) {
+            case NO_OP:
```

Comment:
We don't need to handle `NO_OP` separately. NoOpCompressor is a passthrough compressor. Also, It's better to still record a compressed record value for the sake of consistency

Reply:
Restructured it a bit to remove special handling for NO_OP